js/src/api/contract/contract.js

```diff
@@ -193,119 +195,71 @@ export default class Contract {
     return func;
   }
 
-  _bindEvent (event) {
-    const subscriptions = [];
-
-    event.subscribe = (_options, callback) => {
-      const subscriptionId = subscriptions.length;
-      const options = Object.assign({}, _options, {
-        address: this._address,
-        topics: [event.signature]
-      });
-
-      this._api.eth
-        .newFilter(options)
-        .then((filterId) => {
-          return this._api.eth
-            .getFilterLogs(filterId)
-            .then((logs) => {
-              callback(this.parseEventLogs(logs));
-
-              subscriptions.push({
-                options,
-                callback,
-                filterId
-              });
-            });
-        });
-
-      return subscriptionId;
+  _bindEvent = (event) => {
+    event.subscribe = (options = {}, callback) => {
```

Comment:
Actually I'm a fan of `subscribe` functions returning an `unsubscribe` functions, so that you don't need to care about any async stuff, or subscriptionsId. It's always just:
```js
componentDidMount() {
  this.off = event.subscribe(() => console.log('Doing something'));
}
componentWillUnmount() {
  this.off();
}
```
It can use promises inside, but from the API user point of view it doesn't really matter.